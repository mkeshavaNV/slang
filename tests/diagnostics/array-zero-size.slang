// array-zero-size.slang

// Test that array size cannot be zero

//DIAGNOSTIC_TEST:SIMPLE:

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    bar();
}

func bar() -> int[0];
