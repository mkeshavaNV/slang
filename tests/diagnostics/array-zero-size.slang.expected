result code = -1
standard error = {
tests/diagnostics/array-zero-size.slang(13): error 30025: array size must be larger than zero.
func bar() -> int[0];
                  ^
}
standard output = {
}